var/spack/repos/builtin/packages/bcl2fastq2/package.py

```diff
@@ -0,0 +1,91 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import os
+from spack.util.executable import which
+import shutil
+import llnl.util.tty as tty
+
+
+#class Bcl2fastq2(CMakePackage):
+class Bcl2fastq2(Package):
+    """The bcl2fastq2 Conversion Software converts base
+       call (BCL) files from a sequencing run into FASTQ
+       files."""
+
+    homepage = "https://support.illumina.com/downloads/bcl2fastq-conversion-software-v2-18.html"
+    url      = "https://support.illumina.com/content/dam/illumina-support/documents/downloads/software/bcl2fastq/bcl2fastq2-v2-18-0-12-tar.zip"
+
+    version('2-18-0-12', 'fbe06492117f65609c41be0c27e3215c')
+
+    depends_on('boost@1.54.0')
+    depends_on('cmake@2.8.9:')
+    depends_on('libxml2@2.7.8')
+    depends_on('libxslt@1.1.26~crypto')
+    depends_on('libgcrypt')
+    depends_on('zlib')
+
+    # They forgot to set the flag when they find a library that makes
+    # them happy.
+    patch('cmake-macros.patch')
+    # After finding the libxslt bits, still need to wire in the libexslt
+    # bits.
+    patch('cxxConfigure-cmake.patch')
+
+    root_cmakelists_dir = '../src'
+
+    def url_for_version(self, version):
+        url = "https://support.illumina.com/content/dam/illumina-support/documents/downloads/software/bcl2fastq/bcl2fastq2-v{0}-tar.zip"
+        return url.format(version.dashed)
```

Comment:
I took a stab at making this URL work but it broke so many other packages that I think I'm going to leave things as is for now. Over time, as our sample size increases, it might be a good idea to revisit this URL.